site/content/integrate/apps/quickstart/quick-start-ts/_index.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Hello world in TypeScript"
+heading: "Create a hello world Mattermost App in TypeScript"
+description: "This quick start guide will walk you through the basics of creating a hello world Mattermost App in TypeScript."
+weight: 10
+aliases:
+  - /integrate/apps/quick-start-js/
+---
+
+This quick start guide will walk you through the basics of creating a hello world Mattermost App in TypeScript. In this guide you will review an App that:
+
+- Contains a `manifest.json`, declares itself an HTTP application that acts as a bot, and attaches to locations in the user interface.
+- Contains a `form` with a `submit` function that can launch a modal (if applicable) and send an interpolated message back to the user.
+- Attaches an icon button to the channel header and creates a `/node-example` slash command to provide functionality.
+
+## Prerequisites
+
+Before you can start with your App, you first need to set up your environment by following the [developer setup guide]({{< ref "/integrate/apps/quickstart" >}}).
+
+You also need [NodeJS v16 (Gallium) LTS](https://nodejs.org/en/download/).
```

Comment:
I don't know what range of versions the App framework works with. [Gallium is the current Active LTS for Node](https://nodejs.org/en/about/releases/) so I always default to it. @neflyte pointed out that we don't want to just say "LTS" because there are older LTS versions that probably don't work.

Reply:
@azigler I still think we should de-emphasize and just say:

> Download the LTS or current version of Node [here](https://nodejs.org/en)

That link gives only the latest relevant versions for Node.

My concern is that many people will think they need that exact version of Node that is currently mentioned above. Some will upgrade what they currently have, and some will downgrade what they currently have, when in reality both scenarios could have been using a valid version that works with our software here.

Saying "download this exact version of x" can be seen as a red flag of compatibility, and really just gives me frustration as a dev that I need to configure something in a specific way for something I'm working on. We already have this actual issue with plugin projects so I know the burden of this, and having to explain to individual people that they need to use a specific version of Node kind of sucks for both parties (and I of course feel the judgment of everyone watching the conversation about _having_ to explain that, due to non-updated projects our team owns).

In the case of the App framework docs, I think we should be vague about this version, since we don't have a reason to say why any version is not compatible. As long as our JS libraries (which are pretty minimal) are compatible with what's in that link, it should work fine.

Reply:
I suggest we select a minimum-compatible version of NodeJS and stick with it. If that's Node v14 or v16, it doesn't matter; we can always update this in the future.

The downside of not selecting a version - but not necessarily a point release - is developers who run into dependency issues and then have to reach out for assistance. I realize avoiding dependency issues in JS is fraught with peril, however I feel we should do the best we can to cushion implementers against issues that we know about.